examples/example-hostname/src/main/java/io/grpc/examples/hostname/HostnameServer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.hostname;
+
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.health.v1.HealthCheckResponse.ServingStatus;
+import io.grpc.protobuf.services.ProtoReflectionService;
+import io.grpc.services.HealthStatusManager;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A server that hosts HostnameGreeter, plus infrastructure services like health and reflection.
+ *
+ * <p>This server is intended to be a general purpose "dummy" server.
+ */
+public final class HostnameServer {
+  public static void main(String[] args) throws IOException, InterruptedException {
+    int port = 50051;
+    String hostname = null;
+    if (args.length >= 1) {
+      try {
+        port = Integer.parseInt(args[0]);
+      } catch (NumberFormatException ex) {
+        System.err.println("Usage: [port [hostname]]");
```

Comment:
Make `port` and `hostname` flags?

I think flags are easier to use, and it can use default port but override the hostname.

Reply:
We don't use flags in java OSS.

Reply:
OK...

Do we want to also not use flags in the other languages, to be consistent?

Reply:
The problem is that Java lacks basic things like a flag parsing library. So I'd get to implement the parser. For an example, that seemed a bit much as it makes the example harder to understand. I guess there is also the issue that different languages handle flags differently; I don't know if we want to keep the command-line arguments the same cross-language. I don't see it being a big deal to repeat the port if you want to specify the hostname.